File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int currSum=0;
        int mxSum=INT_MIN;
        for(auto it:nums){
            currSum+=it;
            mxSum=max(mxSum,currSum);
            if(currSum<0){
                currSum=0;
            }
            
        }
        return mxSum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxi=Integer.MIN_VALUE;
        int len=nums.length;

        int sum=0;

        for(int i=0;i<len;i++){
            sum+=nums[i];
            maxi=Math.max(maxi,sum);
            if(sum<0) sum=0;
        }
        return maxi;
        
    }
}